### PR TITLE
Add trapezoidal profile drivetrain controller.

### PR DESCRIPTION
**Pull request by PhilipAxelrod**
Adds drivetrain controller with the following behaviour. A constant acceleration to a cruising velocity, followed by maintaining a cruising velocity, followed by a constant decceleration, getting to the target position at the target final velocity.

https://app.asana.com/0/219272915179160/250106024468357

### CODE REVIEW

**Review by kunalsheth — APPROVED** [review]
cool

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleMotionProfileControllers.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      val accelerationValue = acceleration.toFeetPerSecondSquared
+
+      FeetPerSecond(
+        sqrt(
+          abs(
+            finalVelocitySquared + 2 * -accelerationValue * errorValue)))
+    }
+
+    velocityDeccel.zip(velocityAccel).zip(signError).map {
+      case ((velDec, velAcc), sign) => sign * Utility.min(
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleMotionProfileControllers.scala]: Instead of using `Utility`, can you just do `velDec min crusingVelocity min velAcc`?

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleMotionProfileControllers.scala [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+trait UnicycleMotionProfileControllers[DriveSignal, DrivetrainHardware <: UnicycleHardware, DrivetrainProperties <: UnicycleProperties]
+  extends UnicycleCoreControllers[DriveSignal, DrivetrainHardware, DrivetrainProperties] {
+
+  /**
+    * This controller uses a control loop that results in constant acceleration,
+    * then a constant cruising velocity, and then a constant deceleration to the
+    *  target final velocity at the target position. It is meant to control
+    *  second order systems.
+    * @param initVelocity in feetPerSecond
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleMotionProfileControllers.scala]: Because we no longer care about the actual value units (we only care that they are the correct quantity type), can you remove the annotations about the units of each param?

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala [hunk 3]
```diff
@@ -52,53 +52,15 @@ trait UnicycleDrive extends Drive {
...
-    def closedLoopControl(unicycle: SignalLike[UnicycleSignal])
-                         (implicit hardware: DrivetrainHardware,
-                          props: DrivetrainProperties): PeriodicSignal[UnicycleSignal] = {
-      import props._
-
-      velocityControl(unicycle.map(s => UnicycleVelocity(
-        maxForwardVelocity * s.forward, maxTurnVelocity * s.turn
-      )))
+  object UnicycleControllers extends UnicycleCoreControllers[DriveSignal, DrivetrainHardware, DrivetrainProperties]
+    with UnicycleMotionProfileControllers {
```
Comment (shadaj) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala]: This needs to have the same type params as `UnicycleCoreControllers`. I'll improve this in a later PR.

**Review by shadaj — CHANGES_REQUESTED** [review]
Almost perfect! Just a couple small changes needed (+ testing of course).

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/TwoSidedDrive.scala [hunk 4]
```diff
@@ -49,6 +49,7 @@ trait TwoSidedDrive extends UnicycleDrive { self =>
   protected implicit val clock: Clock
   protected val updatePeriod: Time
 
+  //TODO: add docs
```
Comment (PhilipAxelrod) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/TwoSidedDrive.scala:52]: Oops, I don't know why that's there

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala [hunk 5]
```diff
@@ -26,13 +26,13 @@ trait UnicycleHardware {
   val turnVelocity: Signal[AngularVelocity]
 }
 
+case class UnicycleSignal(forward: Dimensionless, turn: Dimensionless)
```
Comment (PhilipAxelrod) [inline commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala:29]: Wait, why is this here?

File: commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala (hunk 3 above)
Comment (PhilipAxelrod) [reply commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala]: Btw, this doesn't compile:
Error:(56, 10) trait UnicycleMotionProfileControllers takes type parameters
    with UnicycleMotionProfileControllers {
         ^
When I have it take type parameters. I get these nice error in UnicycleDriveTest

Error:(85, 38) value UnicycleVelocity is not a member of UnicycleDriveTest.this.TestDrivetrain
      val in = Signal.constant(drive.UnicycleVelocity(MetersPerSecond(fwd), DegreesPerSecond(turn)))
                                     ^

Comment (PhilipAxelrod) [reply commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleDrive.scala]: By removing the drive, this fixes things. Don't know if you want things that way though

**Review by shadaj — APPROVED** [review]
Let's do this merge thing!
